opts/ip_test.go

```diff
@@ -0,0 +1,42 @@
+package opts
+
+import (
+	"net"
+	"testing"
+)
+
+func TestNewIpOptInvalidDefaultVal(t *testing.T) {
+	ip := net.IPv4(127, 0, 0, 1)
+	defaultVal := "Not an ip"
+
+	ipOpt := NewIpOpt(&ip, defaultVal)
+
+	expected := "127.0.0.1"
+	if ipOpt.String() != expected {
+		t.Fatalf("Expected [%v], got [%v]", expected, ipOpt.String())
+	}
+}
+
+func TestNewIpOptValidDefaultVal(t *testing.T) {
+	ip := net.IPv4(127, 0, 0, 1)
+	defaultVal := "192.168.1.1"
+
+	ipOpt := NewIpOpt(&ip, defaultVal)
```

Comment:
not really a comment on the test, but I just don't get all of the levels of indirection in this IpOpt stuff.  Maybe I'm missing something but why not just have it be `NewIpOpt(defaultValue)`??  What's the point of passing in the 'ip' if all we're going to do is try to override it with 'defaultValue' ?  @LK4D4 do you know the history here?